cmd/kube-apiserver/app/options/validation.go

```diff
@@ -127,6 +128,23 @@ func validateTokenRequest(options *ServerRunOptions) []error {
 	return errs
 }
 
+func validateAPIPriorityAndFairness(options *ServerRunOptions) []error {
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.APIPriorityAndFairness) && options.GenericServerRunOptions.EnablePriorityAndFairness {
+		// we know we need the alpha API enabled.  There are only a few ways to turn it on
+		enabledAPIString := options.APIEnablement.RuntimeConfig.String()
+		switch {
+		case strings.Contains(enabledAPIString, "api/alpha=true"):
+			return nil
+		case strings.Contains(enabledAPIString, "flowcontrol.apiserver.k8s.io/v1alpha1=true"):
+			return nil
+		default:
+			return []error{fmt.Errorf("enabling APIPriorityAndFairness requires --runtime-confg=flowcontrol.apiserver.k8s.io/v1alpha1=true to enable the required API")}
```

Comment:
Should we add the reverse check, too?

Reply:
no? requiring the feature to be on if the API is enabled would break anything that runs with `api/all=true`